shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.h

```diff
@@ -0,0 +1,83 @@
+/// This delegate is used to create AccessibilityBridge in the macOS embedding.
+class AccessibilityBridgeMacDelegate : public AccessibilityBridge::AccessibilityBridgeDelegate {
+ public:
+  //---------------------------------------------------------------------------
+  /// @brief      Creates an AccessibilityBridgeMacDelegate.
+  /// @param[in]  flutterEngine     The weak reference to the FlutterEngine.
+  explicit AccessibilityBridgeMacDelegate(__weak FlutterEngine* flutter_engine);
+  virtual ~AccessibilityBridgeMacDelegate() = default;
+
+  // |AccessibilityBridge::AccessibilityBridgeDelegate|
+  void OnAccessibilityEvent(ui::AXEventGenerator::TargetedEvent targeted_event) override;
+
+  // |AccessibilityBridge::AccessibilityBridgeDelegate|
+  void DispatchAccessibilityAction(AccessibilityNodeId target,
+                                   FlutterSemanticsAction action,
+                                   const std::vector<uint8_t>& data) override;
+
+  // |AccessibilityBridge::AccessibilityBridgeDelegate|
+  std::unique_ptr<FlutterPlatformNodeDelegate> CreateFlutterPlatformNodeDelegate() override;
+
+ private:
+  /// A wrapper structure to wraps macOS native accessibility events.
+  struct NSAccessibilityEvent {
+    NSAccessibilityNotificationName name;
+    gfx::NativeViewAccessible target;
+    NSDictionary* user_info;
+  };
+
+  //---------------------------------------------------------------------------
+  /// @brief      Posts the given event against the given node to the macOS
+  ///             accessibility notification system.
+  /// @param[in]  native_node       The event target, must not be nil.
+  /// @param[in]  mac_notification  The event name, must not be nil.
+  void FireNativeMacNotification(gfx::NativeViewAccessible native_node,
+                                 NSAccessibilityNotificationName mac_notification);
+
+  //---------------------------------------------------------------------------
+  /// @brief      Posts the given event against the given node with the
+  ///             additional attributes to the macOS accessibility notification
+  ///             system.
+  /// @param[in]  native_node       The event target, must not be nil.
+  /// @param[in]  mac_notification  The event name, must not be nil.
+  /// @param[in]  user_info         The additional attributes, must not be nil.
+  void FireNativeMacNotificationWithUserInfo(gfx::NativeViewAccessible native_node,
+                                             NSAccessibilityNotificationName mac_notification,
+                                             NSDictionary* user_info);
+
+  //---------------------------------------------------------------------------
+  /// @brief      Whether the given event is in current pending events.
+  /// @param[in]  event_type        The event you would like to look up.
+  bool IsInGeneratedEventBatch(ui::AXEventGenerator::Event event_type) const;
```

Comment:
Would `bool HasPendingEvent` be more accurate?

Why `event_type` rather than `event`?